lib/trino-parquet/src/test/java/io/trino/parquet/crypto/TestParquetEncryption.java

```diff
@@ -0,0 +1,611 @@
+            "message doc { required int32 id; required int32 age; }");
+
+    // ────────────────────────────────  PARAMETERISED TESTS  ─────────────────────────────
+
+    /**
+     * Column encryption only – footer left in plaintext.
+     */
+    @ParameterizedTest(name = "columnOnly_footerIntegrity={0}_compressed={1}")
+    @CsvSource({
+            "false,false",
+            "false,true",
+            "true,false",
+            "true,true"
+    })
+    void columnOnlyFooterPlaintext(boolean checkFooterIntegrity, boolean compressed)
+            throws IOException
+    {
+        File file = Files.createTempFile("pme‑col‑only", ".parquet").toFile();
+        file.deleteOnExit();
+
+        writeSingleColumnFile(file, compressed, /*encryptFooter*/ false);
+
+        List<Integer> values = readSingleColumnFile(
+                file,
+                new TestingKeyRetriever(checkFooterIntegrity ? Optional.of(KEY_FOOT) : Optional.empty(), Optional.of(KEY_AGE), Optional.empty()),
+                checkFooterIntegrity,
+                Optional.empty());
+
+        verifySequence(values, 100);
+    }
+
+    /**
+     * Column+ footer encryption (same column as above).
+     */
+    @ParameterizedTest(name = "column+footer_compressed={0}")
+    @CsvSource({"false", "true"})
+    void columnAndFooter(boolean compressed)
+            throws IOException
+    {
+        File file = Files.createTempFile("pme‑col‑foot", ".parquet").toFile();
+        file.deleteOnExit();
+
+        writeSingleColumnFile(file, compressed, /*encryptFooter*/ true);
+
+        List<Integer> values = readSingleColumnFile(
+                file,
+                new TestingKeyRetriever(Optional.of(KEY_FOOT), Optional.of(KEY_AGE), Optional.empty()),
+                /*checkFooterIntegrity*/ true,
+                Optional.empty());
+
+        verifySequence(values, 100);
+    }
+
+    /**
+     * Single column encrypted with footer column.
+     */
+    @ParameterizedTest(name = "columnEncryptedWithFooterKey+footer_encrypted={0}")
+    @CsvSource({"false", "true"})
+    void columnEncryptedWithFooterKey(boolean encryptFooter)
+            throws IOException
+    {
+        File file = Files.createTempFile("pme‑footerKey‑col", ".parquet").toFile();
+        file.deleteOnExit();
+
+        // write: column encrypted with footer key, footer encrypted as usual
+        writeSingleColumnFile(file, /*compressed*/ false, encryptFooter, /*columnEncryptedWithFooterKey*/ true, Optional.empty(), () -> range(0, 100));
+
+        // read: supply ONLY the footer key (no column key necessary)
+        List<Integer> values = readSingleColumnFile(
+                file,
+                new TestingKeyRetriever(Optional.of(KEY_FOOT), Optional.empty(), Optional.empty()),
+                /*checkFooterIntegrity*/ true,
+                Optional.empty());
+
+        verifySequence(values, 100);   // data round‑trip OK
```

Comment:
same comment (we also need to verify that an encrypted file cannot be read without encryption keys. e.g. a test where the footer key is not provided to the reader, resulting in an exception.)